src/Models/Context/OatPageContext/OatPageContext.tsx

```diff
@@ -0,0 +1,610 @@
+                        createProject(name, namespace, draft);
+                    }
+                } else {
+                    logDebugConsole(
+                        'warn',
+                        `Could not find project to delete with id: ${action.payload.id}`
+                    );
+                }
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.DUPLICATE_PROJECT: {
+                const id = createGUID();
+                // duplicate the project
+                const project = convertStateToProject(draft);
+                project.projectName =
+                    project.projectName +
+                    '-' +
+                    i18n.t('OATCommon.duplicateFileNameSuffix');
+                draft.ontologyFiles.push({ id: id, data: project });
+
+                // save data
+                saveData(draft);
+
+                // switch to the new project
+                switchCurrentProject(id, draft);
+
+                logDebugConsole(
+                    'debug',
+                    `Created new project with id: ${id}, {project}`,
+                    project
+                );
+                break;
+            }
+            case OatPageContextActionType.SWITCH_CURRENT_PROJECT: {
+                switchCurrentProject(action.payload.projectId, draft);
+                break;
+            }
+            case OatPageContextActionType.SET_CURRENT_PROJECT_NAME: {
+                draft.currentOntologyProjectName = action.payload.name || '';
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.SET_CURRENT_TEMPLATES: {
+                draft.currentOntologyTemplates = action.payload.templates || [];
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.SET_CURRENT_MODELS: {
+                draft.currentOntologyModels = action.payload.models || [];
+                setSelectedModel(draft.selection, draft);
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.SET_CURRENT_MODELS_METADATA: {
+                draft.currentOntologyModelMetadata =
+                    action.payload.metadata || [];
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.SET_CURRENT_MODELS_POSITIONS: {
+                draft.currentOntologyModelPositions =
+                    action.payload.positions || [];
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.SET_CURRENT_NAMESPACE: {
+                draft.currentOntologyNamespace = action.payload.namespace || '';
+                saveData(draft);
+                break;
+            }
+            case OatPageContextActionType.SET_CURRENT_PROJECT: {
+                // TODO: Deprecate in favor of setting via id
```

Comment:
Will this TODO be done in another PR?

Reply:
ehhh. probably not. I'll just remove the comment. I doubt I'll get to actually do this.